presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -0,0 +1,254 @@
+                            partitionUpdate.getName(),
+                            fileNames.size(),
+                            fileSizes.size()));
+        }
+
+        // Compress the file names into a consolidated string
+        partitionMetadata.put(FILE_NAMES, compressFileNames(fileNames));
+
+        // Compress the file sizes
+        partitionMetadata.put(FILE_SIZES, compressFileSizes(fileSizes));
+
+        partitionMetadata.put(MANIFEST_VERSION, VERSION_1);
+        partitionMetadata.putAll(metadata);
+
+        return partitionMetadata.build();
+    }
+
+    public static OptionalLong getManifestSizeInBytes(ConnectorSession session, PartitionUpdate partitionUpdate, Map<String, String> parameters)
+    {
+        if (isFileRenamingEnabled(session) && partitionUpdate.containsNumberedFileNames()) {
+            if (parameters.containsKey(MANIFEST_VERSION)) {
+                return OptionalLong.of(parameters.get(FILE_NAMES).length() + parameters.get(FILE_SIZES).length());
+            }
+            List<FileWriteInfo> fileWriteInfos = partitionUpdate.getFileWriteInfos();
+            return OptionalLong.of(compressFileNames(fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList())).length()
+                    + compressFileSizes(fileWriteInfos.stream().map(FileWriteInfo::getFileSize).filter(Optional::isPresent).map(Optional::get).collect(toImmutableList())).length());
+        }
+
+        return OptionalLong.empty();
+    }
+
+    static String compressFileNames(List<String> fileNames)
+    {
+        if (fileNames.size() == 1) {
+            return fileNames.get(0);
+        }
+
+        boolean isContinuousSequence = true;
+        int start = 0;
+        for (String name : fileNames) {
+            if (start != Integer.parseInt(name)) {
+                isContinuousSequence = false;
+                break;
+            }
+            start++;
+        }
+
+        if (isContinuousSequence) {
+            return fileNames.get(fileNames.size() - 1);
+        }
+
+        return compressFileNamesUsingRoaringBitmap(fileNames);
+    }
+
+    static List<String> decompressFileNames(String compressedFileNames)
+    {
+        // Check if the compressed fileNames string is a number
+        if (compressedFileNames.matches("\\d+")) {
+            long end = Long.parseLong(compressedFileNames);
+
+            if (end == 0) {
+                return ImmutableList.of("0");
+            }
+
+            return LongStream.range(0, end + 1).mapToObj(String::valueOf).collect(toImmutableList());
+        }
+
+        try {
+            RoaringBitmap roaringBitmap = new RoaringBitmap();
+            ByteBuffer byteBuffer = ByteBuffer.wrap(compressedFileNames.getBytes(ISO_8859_1));
+            roaringBitmap.deserialize(byteBuffer);
+            return Arrays.stream(roaringBitmap.toArray()).mapToObj(Integer::toString).collect(toImmutableList());
+        }
+        catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)